File: 1주차/송성욱/카펫.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int carpet = yellow + brown;

    //sqrt 제곱근 반환 함수사용 생각못함

    for (int height = 3; height <= sqrt(carpet); height++)
    {
        if (carpet % height == 0)
        {
            int width = carpet / height;


            if (((width + height) * 2 - 4) == brown)
            {
                answer.push_back(width);
                answer.push_back(height);
                break;
            }

        }


    }


    return answer;
}
//예제 입력
//int main() 
//{
//   
//
//   
//    int brown = 10;
//    int yellow = 2;
//
//   
//    vector<int> result = solution(brown, yellow);
//
//    cout << "Width: " << result[0] << ", Height: " << result[1] << endl;
//
//    return 0;
//}

File: 1주차/김용상/카펫.java
import java.lang.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        //감싸는 공식 존재. yellow 기준 1x24 => brown = 54개 .. 반복
        for (int i=1;i<Math.sqrt(yellow)+1;i++){
            if (yellow%i!=0) continue;
            int x = yellow/i;  // 항상 x>=y
            int y = i;
            if (brown==((2*x)+(y*2)+4)){
                return new int[]{x+2,y+2};
            }
        }
        return answer;
    }
}
